### ISSUE TITLE
Webpack Compilation Error when module is not imported

### ISSUE THREAD

**Issue by amitguptagwl** [cypress-io/cypress#17985]
### Current behavior

Cypress throws `Webpack Compilation Error` when running following test file

```js
const { fakeServer } = require("cypress/types/sinon")

describe('example to-do app', () => {
  it.only('displays two todo items by default', () => {
    fs.readFile("fileDoesn'tExist.js");
  });
});
```

```bash
Error: Webpack Compilation Error
./cypress/integration/1-getting-started/todo.spec.js
Module not found: Error: Can't resolve 'cypress/types/sinon' in '/home/amit/temp/cytorus_code/test-repository/test/cypress/integration/1-getting-started'
resolve 'cypress/types/sinon' in '/home/amit/temp/cytorus_code/test-repository/test/cypress/integration/1-getting-started'
  Parsed request is a module
  using description file: /home/amit/temp/cytorus_code/test-repository/package.json (relative path: ./test/cypress/integration/1-getting-started)
    Field 'browser' doesn't contain a valid alias configuration
    Looked for and couldn't find the file at the following paths:

```

### Desired behavior

I'm expecting it to report the error in particular test with line number, if possible.

### Test code to reproduce

You can reproduce this error with following code anywhere in the test

```js
const { fakeServer } = require("cypress/types/sinon")

describe('example to-do app', () => {
  it('throws Webpack Compilation Error', () => {
    fs.readFile("fileDoesn'tExist.js"); //sorry-cypress not working
  });
});
```
or
```js
const { fakeServer } = require("cypress/types/sinon")

describe('example to-do app', () => {
  it('throws Webpack Compilation Error', () => {
     cy.then( () => {
       throw new Error("on my wish");
     })
  });
});
```

### Cypress Version

8.0.0, 8.3.1

**Comment by lmiller1990** [comment]
`const { fakeServer } = require("cypress/types/sinon")` is just type definitions - there's no actual code (like a `fakeServer` function) to import.

You might want to be importing from `sinon` directly, perhaps.

Cypress will do its best to show you the location and stack trace from an error out of the box - you shouldn't need to import anything (like `fakeServer`).

**Comment by amitguptagwl** [comment]
@lmiller1990 this is the sample code I took from the generated test code by Cypress only. Nothing is added by me. However, the problem is not with `fakeServer`. Even if I remove that line I get "Webpack Compilation Error". The problem is when an exception is thrown inside a promise or when some module is not imported.

**Comment by lmiller1990** [comment]
Can you share a repository reproducing the problem?

**Comment by amitguptagwl** [comment]
Sure I'll create a repo and share the link

**Comment by amitguptagwl** [comment]
I found that this issue has been resolved in Cypress v8.4.1
